//
// Copyright 2020 BigQuery Utils
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package bigquery.utils.zetasql_helper;

import "zetasql/public/parse_location_range.proto";
import "zetasql/public/value.proto";

option java_package = "com.google.bigquery.utils.zetasqlhelper";

message ParseTokenProto {

  enum Kind {
    KEYWORD = 0;                // A zetasql keyword or symbol.
    IDENTIFIER = 1;             // An identifier that was quoted.
    IDENTIFIER_OR_KEYWORD = 2;  // An unquoted identifier.
    VALUE = 3;                  // A literal value.
    COMMENT = 4;                // A comment.
    END_OF_INPUT = 5;           // The end of the input string was reached.
  };

  optional string image = 1;
  optional Kind kind = 2;
  optional zetasql.ParseLocationRangeProto parse_location_range = 3;
  optional zetasql.ValueProto value = 4;
}
